src/routes/transactions/entities/swap-order-info.entity.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  TransactionInfo,
+  TransactionInfoType,
+} from '@/routes/transactions/entities/transaction-info.entity';
+import {
+  ApiExtraModels,
+  ApiProperty,
+  ApiPropertyOptional,
+} from '@nestjs/swagger';
+
+export class TokenInfo {
+  @ApiPropertyOptional()
+  logo: string | null;
+
+  @ApiProperty()
+  symbol: string;
+
+  @ApiProperty()
+  amount: string;
+
+  constructor(args: { logo: string | null; symbol: string; amount: string }) {
+    this.logo = args.logo;
+    this.symbol = args.symbol;
+    this.amount = args.amount;
+  }
+}
+
+@ApiExtraModels(TokenInfo)
+export abstract class SwapOrderTransactionInfo extends TransactionInfo {
+  @ApiProperty()
+  status: 'open' | 'fulfilled' | 'cancelled' | 'expired';
+
+  @ApiProperty()
+  orderKind: 'buy' | 'sell';
+
+  @ApiProperty()
+  sellToken: TokenInfo;
+
+  @ApiProperty()
+  buyToken: TokenInfo;
+
+  @ApiProperty()
+  expiresTimestamp: number;
+
+  protected constructor(args: {
+    status: 'open' | 'fulfilled' | 'cancelled' | 'expired';
+    orderKind: 'buy' | 'sell';
+    sellToken: TokenInfo;
+    buyToken: TokenInfo;
+    expiresTimestamp: number;
+  }) {
+    super(TransactionInfoType.SwapOrder, null, null);
+    this.status = args.status;
+    this.orderKind = args.orderKind;
+    this.sellToken = args.sellToken;
+    this.buyToken = args.buyToken;
+    this.expiresTimestamp = args.expiresTimestamp;
+  }
+}
+
+@ApiExtraModels(TokenInfo)
+export class FulfilledSwapOrderTransactionInfo extends SwapOrderTransactionInfo {
+  @ApiPropertyOptional()
```

Comment:
```suggestion
  @ApiPropertyOptional({ type: String, nullable: true })
```

Reply:
Changed in [e3a4a9c](https://github.com/safe-global/safe-client-gateway/pull/1299/commits/e3a4a9c33aac2d31572dd3cb51e6cacec9ed3515)